db/db_impl_open.cc

```diff
@@ -1217,6 +1218,7 @@ Status DBImpl::Open(const DBOptions& db_options, const std::string& dbname,
   if (s.ok()) {
     ROCKS_LOG_INFO(impl->immutable_db_options_.info_log, "DB pointer %p", impl);
     LogFlush(impl->immutable_db_options_.info_log);
+    impl->FlushWAL(false /*sync*/);
```

Comment:
why flush here?

Reply:
manual_wal_flush is to let user manually flush the changes. The user probably does not feel obliged to do the flush after recovery when no changes is made to the db.  DB thus should start with no buffered log data.

Reply:
I just noticed that recovery does not write anything to the WAL. I still think it would not hurt to keep it just in case a future change does write to the WAL.

Reply:
I would prefer assert WAL being empty and add comment saying WAL needs to flush if the assert does not hold. Or at least add inline comment saying this is just a safeguard.